### PR TITLE
mcstaging/phenoimager2mc

### PR DESCRIPTION
**Pull request by chiarasch**
## PR checklist

Solves this iseeue: #6110

- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR. Test data was added with PR https://github.com/nf-core/test-datasets/pull/1277
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`

Comment: Did not add conda, works with Docker and Singularity which is sufficient for our workflow, as we won't run it with conda.

### CODE REVIEW

**Comment by chiarasch** [comment]
All tests pass except for conda. I do not plan to add conda as it is not required for our pipeline. Please review, thanks!

File: modules/nf-core/mcstaging/phenoimager2mc/meta.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      documentation: "https://github.com/SchapiroLabor/phenoimager2mc/README.md"
+      tool_dev_url: "https://github.com/SchapiroLabor/phenoimager2mc"
+      licence: ["GPL-2.0 license"]
+
+input:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. `[ id:'sample1', single_end:false ]`
```
Comment (migueLib) [inline modules/nf-core/mcstaging/phenoimager2mc/meta.yml:23]: remove single_end:false

File: modules/nf-core/mcstaging/phenoimager2mc/meta.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      description: Folder or list with .tif files of one cycle from PhenoImager
+      #pattern: "*.{bam,cram,sam}"
+
+output:
+  #Only when we have meta
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. `[ id:'sample1', single_end:false ]`
```
Comment (migueLib) [inline modules/nf-core/mcstaging/phenoimager2mc/meta.yml:36]: remove single_end:false

File: modules/nf-core/mcstaging/phenoimager2mc/meta.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  - versions:
+      type: file
+      description: File containing software versions
+      pattern: "versions.yml"
+
+  - tif:
+      type: file
+      description: One output .tif file containing concatenated tiles of the cycle.
+      pattern: "*.{tif}" ## also other formats? ome-tif? tiff?
```
Comment (migueLib) [inline modules/nf-core/mcstaging/phenoimager2mc/meta.yml:46]: add tiff format

File: modules/nf-core/mcstaging/phenoimager2mc/tests/main.nf.test [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    script "../main.nf"
+    process "MCSTAGING_PHENOIMAGER2MC"
+        config "./nextflow.config"
+
+    tag "modules"
+    tag "modules_nfcore"
+    tag "mcstaging"
+    tag "mcstaging/phenoimager2mc"
+
+    test("Should run without failures") {
```
Comment (migueLib) [inline modules/nf-core/mcstaging/phenoimager2mc/tests/main.nf.test:13]: Please use a more descriptive test name  like test("proces - input")

**Review by migueLib — COMMENTED** [review]
Minor comments.

**Comment by SPPearce** [comment]
To stop the tests running on conda, you need to add the [tool to the list here](https://github.com/nf-core/modules/blob/master/.github/workflows/test.yml#L489). Note that there are separate sections for pytest and nf-test, you have to edit the nf-test section!

**Comment by chiarasch** [comment]
Thank you for the review! I added the module to the list but once I push, I get this error: `To https://github.com/chiarasch/nfcore_modules.git
 ! [remote rejected]     mcstaging/phenoimager2mc -> mcstaging/phenoimager2mc (refusing to allow an OAuth App to create or update workflow `.github/workflows/test.yml` without `workflow` scope)
error: failed to push some refs to 'https://github.com/chiarasch/nfcore_modules.git'`
If I change another file (e.g. main.nf in 1f14ebb) I can still push, so I suppose it has sth to do with github workflows? My github token already incudes the workflow scope, do you have an idea?

**Comment by SPPearce** [comment]
> Thank you for the review! I added the module to the list but once I push, I get this error: `To https://github.com/chiarasch/nfcore_modules.git ! [remote rejected] mcstaging/phenoimager2mc -> mcstaging/phenoimager2mc (refusing to allow an OAuth App to create or update workflow `.github/workflows/test.yml`without`workflow` scope) error: failed to push some refs to 'https://github.com/chiarasch/nfcore_modules.git'` If I change another file (e.g. main.nf in [1f14ebb](https://github.com/nf-core/modules/commit/1f14ebb431ad6030d8db756650a5719172f9cbaa)) I can still push, so I suppose it has sth to do with github workflows? My github token already incudes the workflow scope, do you have an idea?

Yes, I've seen that before, but when I try and edit a PR that is coming from someone else's clone of the repo, not sure why you are seeing it here.
The way around it then is to make changes of a branch of this repo, rather than your own fork. That is how I contribute generally (and I use gitpod to make it really easy).

File: modules/nf-core/mcstaging/phenoimager2mc/tests/main.nf.test [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+                    ]
+                ]
+                """
+            }
+        }
+        then {
+            assertAll (
+                { assert process.success },
+                { assert process.out.tif },
+                { assert snapshot(process.out.versions).match("versions") }
```
Comment (SPPearce) [inline modules/nf-core/mcstaging/phenoimager2mc/tests/main.nf.test:32]: Indentation might need altering
```suggestion
                { assert snapshot(
                  file(process.out.tif[0][1]).name,
                  process.out.versions).match() }
```

File: modules/nf-core/mcstaging/phenoimager2mc/meta.yml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. `[ id:'sample1' ]`
+
+  - tiles:
+      type: list
+      description: Folder or list with .tif files of one cycle from PhenoImager
+      #pattern: "*.{bam,cram,sam}"
```
Comment (SPPearce) [inline modules/nf-core/mcstaging/phenoimager2mc/meta.yml:28]:
```suggestion
```

**Review by SPPearce — COMMENTED** [review]
Can you add a stub test too?

File: modules/nf-core/mcstaging/phenoimager2mc/tests/main.nf.test [hunk 7]
```diff
@@ -0,0 +1,65 @@
+nextflow_process {
+
+    name "Test Process MCSTAGING_PHENOIMAGER2MC"
+    script "../main.nf"
+    process "MCSTAGING_PHENOIMAGER2MC"
+        config "./nextflow.config"
```
Comment (kbestak) [inline modules/nf-core/mcstaging/phenoimager2mc/tests/main.nf.test:6]:
```suggestion
    config "./nextflow.config"
```

File: modules/nf-core/mcstaging/phenoimager2mc/tests/main.nf.test [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+                    ]
+                ]
+                """
+            }
+        }
+        then {
+            assertAll (
+                { assert process.success },
+                { assert snapshot(
+                  file(process.out.tif[0][1]).name,
```
Comment (kbestak) [inline modules/nf-core/mcstaging/phenoimager2mc/tests/main.nf.test:32]:
```suggestion
                      file(process.out.tif[0][1]).name,
```

File: modules/nf-core/mcstaging/phenoimager2mc/tests/main.nf.test [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+                  process.out.versions).match() }
```
Comment (kbestak) [inline modules/nf-core/mcstaging/phenoimager2mc/tests/main.nf.test:33]:
```suggestion
                      process.out.versions).match() }
```

File: modules/nf-core/mcstaging/phenoimager2mc/tests/main.nf.test [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+                    ]
+                ]
+                """
+            }
+        }
+        then {
+            assertAll (
+                { assert process.success },
+                { assert snapshot(
+                  file(process.out.tif[0][1]).name,
```
Comment (kbestak) [inline modules/nf-core/mcstaging/phenoimager2mc/tests/main.nf.test:60]:
```suggestion
                      file(process.out.tif[0][1]).name,
```

File: modules/nf-core/mcstaging/phenoimager2mc/tests/main.nf.test [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+                  process.out.versions).match() }
```
Comment (kbestak) [inline modules/nf-core/mcstaging/phenoimager2mc/tests/main.nf.test:61]:
```suggestion
                      process.out.versions).match() }
```

**Review by kbestak — COMMENTED** [review]
Check indentations

**Review by kbestak — APPROVED** [review]
This looks good to me and previous comments have been addressed.

**Pull request merged**